src/widgets/button.rs

```diff
@@ -60,7 +81,16 @@ impl ButtonWidget {
     }
 
     pub fn set_text<S: Into<String>>(&mut self, content: S) {
-        self.content = Some(content.into());
+        let content = Some(content.into());
```

Comment:
I guess that this function should probably always set both `content` and `short_content` to keep existing blocks working as they do now (else, we may have blocks that get displayed with the `short_text` being the `full_text` that was given at initialisation instead of the actual `full_text`)